src/libultra/io/pirawdma.c

```diff
@@ -1,23 +1,23 @@
 #include "global.h"
 
 s32 __osPiRawStartDma(s32 direction, u32 devAddr, void* dramAddr, size_t size) {
```

Comment:
```suggestion
s32 __osPiRawStartDma(s32 direction, uintptr_t devAddr, void* dramAddr, size_t size) {
```
I know you didn't touch this line, but I noticed it needs fixed.

Reply:
What makes the return of this a size_t? I know @Thar0 mentioned there's a lot of stuff in libultra that returns 0 for success and -1 for error.

Reply:
Indeed, this can't be a size_t because of that, surely? If anything it would be an ssize_t

Reply:
It definitely isn't a size_t or similar. In other works where a function returns 0 on success or some negative number as an error indicator, the return type is simply int. Since we don't use int anywhere I think s32 is most appropriate.